TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/transform/SigV4AuthTransformerFactory.java

```diff
@@ -0,0 +1,49 @@
+package org.opensearch.migrations.transform;
+
+import java.nio.ByteBuffer;
+import java.time.Clock;
+import java.util.function.Supplier;
+
+import org.opensearch.migrations.IHttpMessage;
+import org.opensearch.migrations.aws.SigV4Signer;
+import org.opensearch.migrations.replay.datahandlers.http.HttpJsonMessageWithFaultingPayload;
+
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+
+public class SigV4AuthTransformerFactory implements IAuthTransformerFactory {
+    private final AwsCredentialsProvider credentialsProvider;
+    private final String service;
+    private final String region;
+    private final String protocol;
+    private final Supplier<Clock> timestampSupplier;
+
+    public SigV4AuthTransformerFactory(
+            AwsCredentialsProvider credentialsProvider,
+            String service,
+            String region,
+            String protocol,
+            Supplier<Clock> timestampSupplier) {
+        this.credentialsProvider = credentialsProvider;
+        this.service = service;
+        this.region = region;
+        this.protocol = protocol;
+        this.timestampSupplier = timestampSupplier;
+    }
+
+    @Override
+    public IAuthTransformer getAuthTransformer(IHttpMessage httpMessage) {
```

Comment:
This can take HttpJsonMessageWithFaultingPayload and the body should adapt it to an IHttpMessage object.

Reply:
addressed